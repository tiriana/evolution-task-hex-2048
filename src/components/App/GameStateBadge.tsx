import React from "react";

type GameStateProps = {
  hasMove: boolean;
};
export const GameStateBadge: React.FC<GameStateProps> = ({ hasMove }) => {
  const status = hasMove ? "playing" : "game-over";
  return <div data-status={status}>Game status: {status}</div>;
};
