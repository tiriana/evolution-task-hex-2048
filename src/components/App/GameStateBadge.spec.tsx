import React from "react";
import renderer from "react-test-renderer";
import { GameStateBadge } from "./GameStateBadge";

it.each([true, false])("renders game-state", (hasMoves) => {
  const component = renderer.create(<GameStateBadge hasMove={hasMoves} />);

  expect(component.toJSON()).toMatchSnapshot();
});
