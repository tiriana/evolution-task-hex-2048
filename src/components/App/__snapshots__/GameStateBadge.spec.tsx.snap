// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders game-state 1`] = `
<div
  data-status="playing"
>
  Game status: 
  playing
</div>
`;

exports[`renders game-state 2`] = `
<div
  data-status="game-over"
>
  Game status: 
  game-over
</div>
`;
